src/QuikGraph/Helpers/FinallyScope.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Diagnostics;
+using JetBrains.Annotations;
+
+namespace QuikGraph.Utils
+{
+    /// <summary>
+    ///
+    /// </summary>
+    public struct FinallyScope : IDisposable
```

Comment:
If the problem is returning a struct while having an IDisposable return then maybe we can just move to a class rather than stuct, this would allow no boxing.